File: eden/code-samples/eden.go
package main

import "fmt"

/*
  Eden is full of beautiful islands...
*/
type Island struct {
  Name     string
  Length   int
  Width    int
  Features []string
}

func (i *Island) Area() int {
  return i.calcArea()
}

func (i *Island) calcArea() int {
  return i.Length * i.Width 
}

// ...one of them is named Elysian

func main() {
  elysian := &Island{
    Name:     "Elysian",
    Length:   300,
    Width:    350,
    Features: []string{"Sunsets", "Beaches"},
  }

  fmt.Println(elysian.Area())
}

File: eden/code-samples/eden.rb
=begin
Eden is full of beautiful islands...
=end

class Island
  def initialize(opts)
    @name = opts[:name]
    @length = opts[:length]
    @width = opts[:width]
    @features = opts[:features]
  end

  def area
    calcArea
  end

  private

  def calcArea
    @length * @width
  end
end

# ...one of them is named Elysian

elysian = Island.new(
  name: 'Elysian',
  length: 300,
  width: 350,
  features: %w[Sunsets Beaches]
)

puts elysian.area
